scanner/src/main/kotlin/scanners/scancode/ScanCodeResultParser.kt

```diff
@@ -125,14 +136,31 @@ private fun getLicenseFindings(result: JsonNode): List<LicenseFinding> {
     val files = result["files"]?.asSequence().orEmpty()
     files.flatMapTo(licenseFindings) { file ->
         val licenses = file["licenses"]?.asSequence().orEmpty()
-        licenses.map {
+
+        licenses.groupBy(
+            keySelector = {
+                LicenseGrouping(
+                    // Older ScanCode versions do not produce the `license_expression` field.
+                    // Just use the `key` field in this case.
+                    it["matched_rule"]?.get("license_expression")?.textValue() ?: it["key"].textValue(),
+                    it["start_line"].intValue(),
+                    it["end_line"].intValue()
+                )
+            }, valueTransform = {
+                SpdxExpressionReplacement(it["key"].textValue(), getLicenseId(it))
+            }
+        ).map { (licenseGroup, replacements) ->
+            var spdxLicenseExpression = licenseGroup.licenseExpression
+            replacements.forEach {
+                spdxLicenseExpression = spdxLicenseExpression.replace(it.scanCodeLicenseKey, it.spdxExpression)
+            }
```

Comment:
Can be simplified to:

            val spdxLicenseExpression = replacements.fold(licenseGroup.licenseExpression) { expression, replacement ->
                expression.replace(replacement.scanCodeLicenseKey, replacement.spdxExpression)
            }